pkg/spec/deployment.go

```diff
@@ -56,6 +56,9 @@ func (deployment *DeploymentSpecMod) Fix() error {
 
 	deployment.ControllerFields.ObjectMeta.Labels = addKeyValueToMap(appLabelKey, deployment.ControllerFields.Name, deployment.ControllerFields.ObjectMeta.Labels)
 
+	if deployment.ControllerFields.Appversion != "" {
+		deployment.ControllerFields.ObjectMeta.Annotations = addKeyValueToMap("appversion", string(deployment.ControllerFields.Appversion), deployment.ControllerFields.ObjectMeta.Annotations)
```

Comment:
appversion should be a constant. Similarly to [appLabelKey](https://github.com/kedgeproject/kedge/blob/9b732e3d526b03b197c9ba623627099f221f023c/pkg/spec/resources.go#L44)

If you redefine `Appversions` as a string you won't have to do type conversion

Reply:
We should add this label to every object created from the file.
It might be better to do this together for all objects. This could be done at the end of the `CreateK8sObjects` function in the resourece.go , just [before](https://github.com/kedgeproject/kedge/blob/9b732e3d526b03b197c9ba623627099f221f023c/pkg/spec/resources.go#L491) all objects are returned. You might have to do some type conversion magic to get `ObjectMeta` from `runtime.Object` ;-)

Reply:
You also forgot about DeploymentConfigs, this is the reason I think it would be better to do it in CreateK8sObjects so you can be sure you don't forget something.

Reply:
specifically this could be:

```
appVersion
```

Reply:
> just before all objects are returned. You might have to do some type conversion magic to get ObjectMeta from runtime.Object ;-)

@kadel can't we just do this label addition in each `create*` function? Like before we add ObjectMeta in places [[1](https://github.com/kedgeproject/kedge/blob/9b732e3d526b03b197c9ba623627099f221f023c/pkg/spec/resources.go#L259)], [[2](https://github.com/kedgeproject/kedge/blob/9b732e3d526b03b197c9ba623627099f221f023c/pkg/spec/resources.go#L271)], [[3](https://github.com/kedgeproject/kedge/blob/9b732e3d526b03b197c9ba623627099f221f023c/pkg/spec/resources.go#L246)].

Can we have some way where we can enforce the call of function which does addition of this label?